addons/popochiu/engine/objects/graphic_interface/templates/9_verb/9_verb_commands.gd

```diff
@@ -110,7 +110,26 @@ func pull() -> void:
 ## Called when [code]E.current_command == Commands.GIVE[/code] and [code]E.command_fallback()[/code]
 ## is triggered.
 func give() -> void:
-	await C.player.say("What?")
+	if I.active and E.clicked:
+		give_item_to(I.active, E.clicked)
+	elif I.active and I.clicked and I.active != I.clicked:
+		give_item_to(I.active, I.clicked)
+	elif I.clicked:
+		match I.clicked.last_click_button:
+			MOUSE_BUTTON_LEFT:
+				I.clicked.set_active(true)
+			MOUSE_BUTTON_RIGHT:
+				# TODO: I'm not sure this is the right way to do this. Maybe GUIs should capture
+				# 		click inputs on clickables and inventory items. ----------------------------
+				E.current_command = (
+					I.clicked.suggested_command if I.clicked.get("suggested_command")
+					else Commands.LOOK_AT
+				)
+				
+				I.clicked.handle_command(MOUSE_BUTTON_LEFT)
+				# ----------------------------------------------------------------------------------
+	else:	
+		await C.player.say("What?")
```

Comment:
Since this is basically doing the same as `func use()`, what do you think about unifying the logic into a single method? The only thing that would change is the method called in the first two `if` statements. It could be something like:

```
func give() -> void:
	await _give_or_use(give_item_to)


func use() -> void:
	await _give_or_use(use_item_on)


func _give_or_use(callback: Callable) -> void:
	if I.active and E.clicked:
		callback.call(I.active, E.clicked)
	elif I.active and I.clicked and I.active != I.clicked:
		callback.call(I.active, I.clicked)
	# And so on...
```


You don't have to make the change if you don't want to, its just a suggestion.